spring-cloud-bus/src/main/java/org/springframework/cloud/bus/event/RefreshListener.java

```diff
@@ -34,14 +36,38 @@
 
 	private ContextRefresher contextRefresher;
 
+	private ServiceMatcher serviceMatcher;
+
+	@Deprecated
+	// TODO Remove in 3.0.x
 	public RefreshListener(ContextRefresher contextRefresher) {
+		this(contextRefresher, null);
+	}
+
+	public RefreshListener(ContextRefresher contextRefresher,
+			ServiceMatcher serviceMatcher) {
 		this.contextRefresher = contextRefresher;
+		this.serviceMatcher = serviceMatcher;
 	}
 
 	@Override
 	public void onApplicationEvent(RefreshRemoteApplicationEvent event) {
-		Set<String> keys = this.contextRefresher.refresh();
-		log.info("Received remote refresh request. Keys refreshed " + keys);
+		log.info("Received remote refresh request.");
+		// TODO Remove this in 3.0.x
+		if (serviceMatcher == null) {
```

Comment:
I think if `serviceMatcher == null` we need to preserve existing behavior and do the refresh? Maybe still log the warning. What do you think?

Reply:
Yeah good point.

Reply:
I doubt there are many users that create their own `RefreshListener`, but for the one team out there :-)

Reply:
Someone will prove us wrong....should be all set now!